### PR TITLE
Added Interact type, which funnels in Pipes and Control into one type.

### PR DESCRIPTION
**Pull request by elisarver**
This was mainly a re-framing of what Pipes and the command do, broadly, returning it as a combination two short interfaces.

### CODE REVIEW

File: pipes.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Pipes Returns the raw types object, because
+// this may be something we want to make by scratch
+// in a future type, such as in situations where you
+// filter/wrap output.
+func (p *Pipes) Pipes() *Pipes {
+	return p
+}
+
+// StdoutScanner returns a bufio.Scanner over stdio.
+func (p *Pipes) StdoutScanner() *bufio.Scanner {
```
Comment (dpastoor) [inline pipes.go]: would it be more appropriate to call this `NewStdoutScanner()`?

My first experiment with the interface without reading the source code carefully I thought this was just returning an already initialized Scanner so i did...

```
for p.StdoutScanner().Scan() {
			text := p.StdoutScanner().Text()
   ...
```

**Review by dpastoor — COMMENTED** [review]
experimenting with the interface will add some comments as I go!

File: pipes.go (hunk 1 above)
Comment (elisarver) [reply pipes.go]: `*(exec.Cmd).StdoutPipe()` is the pattern I'm following, which *also* creates a new thing.

In both cases, having two copies would make no sense. So my fix would probably be to add two private sync.RunOnce types, and returning error for the second request.

Comment (dpastoor) [reply pipes.go]: gotcha, would the interface then change to *bufioScanner, error I guess?

yeah i can't decide if that would be better, or just adding a little more in the docstring. like

the most verbose I could think of would be something like

"StdoutScanner returns a newly initialized bufio.Scanner over stdio, should only be called once"

**Comment by dpastoor** [comment]
nice I like that single initiation now 👍

**Pull request merged**